dashboard/app/controllers/passwords_controller.rb

```diff
@@ -4,6 +4,19 @@ class PasswordsController < Devise::PasswordsController
 
   append_after_action :show_reset_url_if_admin, only: :create
 
+  def create
+    email = request.parameters['user']['email']
+    if email_in_hoc_2016_signups?(email)
+      # If the user has a full account as well, don't use the HOC flow
+      user = User.find_by_email_or_hashed_email(email)
+      unless user
+        redirect_to '/users/sign_up?already_hoc_registered=true'
```

Comment:
Unfortunately, this did not work, the parameter is getting swallowed. A brief investigation didn't turn up any good solution, so I replaced this with:

redirect_to "#{new_user_registration_path}?already_hoc_registered=true"

which at least uses the new_user_registration_path. I'll look into this later.